app/src/lib/git/diff.ts

```diff
@@ -54,24 +109,49 @@ export function getWorkingDirectoryDiff(repository: Repository, file: WorkingDir
     //
     // citation in source:
     // https://github.com/git/git/blob/1f66975deb8402131fbf7c14330d0c7cdebaeaa2/diff-no-index.c#L300
-    opts = { successExitCodes: new Set([ 0, 1 ]) }
-    args = [ 'diff', '--no-ext-diff', '--no-index', '--patch-with-raw', '-z', '--no-color', '--', '/dev/null', file.path ]
+    successExitCodes = new Set([ 0, 1 ])
+
+    args = [ 'diff', '--no-ext-diff', '--no-index', '--patch-with-raw', '-z', '--no-color' ]
+
+    // enable --binary unless we know it's an image format
+    const extension = Path.extname(file.path)
+    if (!imageFileExtensions.has(extension)) {
+      args.push('--binary')
```

Comment:
Nevermind, this isn't important to the PR - I just need to force node to output binary content.

There's still something weird up with the diff changes here that I need to get to the bottom of though :hurtrealbad: